src/main/java/net/fabricmc/loom/configuration/providers/forge/ForgeRunTemplate.java

```diff
@@ -58,6 +58,9 @@ public void applyTo(RunConfigSettings settings, ConfigValue.Resolver configValue
 
 		settings.vmArgs(CollectionUtil.map(jvmArgs, value -> value.resolve(configValueResolver)));
 
+		// Add MOD_CLASSES, this is something that ForgeGradle does
+		env.computeIfAbsent("MOD_CLASSES", $ -> ConfigValue.of("{source_roots}"));
```

Comment:
`env` is not necessarily mutable I think, you could add it to `settings.getEnvironmentVariables` directly